### ISSUE TITLE
JSON exported multiple points for the same object

### ISSUE THREAD

**Issue by lavalex** [ibm-messaging/mq-metric-samples#188]
Hello,

Why there are multiple points for the same object? 
For example, there is point with type "qmgr" and type "queueManager"? Same happens with queues ... There are two entries for the same queue with different metrics/attributes.

Also, queues get two points with different set of metrics in the same JSON file. Is it possible to group them into one?

Thanks.

Example (from the same JSON file):

`{
			"objectType": "qmgr",
			"tags": {
				"platform": "UNIX",
				"qmgr": "JMSQM"
			},
			"metrics": {
				"channelInitiatorStatus": 2,
				"commandServerStatus": 2,
				"connectionCount": 26,
				"status": 2,
				"uptime": 73994
			}
		}`

`{
      "objectType": "queueManager",
      "tags": {
        "platform": "UNIX",
        "qmgr": "JMSQM"
      },
      "metrics": {
        "alterDurableSubscriptionCount": 0,
        "commitCount": 0,
        "concurrentConnectionsHighWaterMark": 26,
        "cpuLoadFifteenMinuteAveragePercentage": 0,
        "cpuLoadFiveMinuteAveragePercentage": 0.02,
        "cpuLoadOneMinuteAveragePercentage": 0.05,
        "createDurableSubscriptionCount": 0,
        "createNonDurableSubscriptionCount": 0,
        "deleteDurableSubscriptionCount": 0,
        "deleteNonDurableSubscriptionCount": 0,
        "durableSubscriberHighWaterMark": 2,
        "durableSubscriberLowWaterMark": 2,
        "expiredMessageCount": 0,
        "exporterPublications": 18,
        "failedBrowseCount": 0,
        "failedCreateAlterResumeSubscriptionCount": 0,
        "failedMqcbCount": 0,
        "failedMqcloseCount": 0,
        "failedMqconnMqconnxCount": 0,
        "failedMqgetCount": 29,
        "failedMqinqCount": 0,
        "failedMqopenCount": 0,
        "failedMqput1Count": 0,
        "failedMqputCount": 0,
        "failedMqsetCount": 0,
        "failedMqsubrqCount": 0,
        "failedTopicMqputMqput1Count": 0,
        "gotNonPersistentMessagesBytes": 73220,
        "gotPersistentMessagesBytes": 0,
        "intervalDestructiveGetTotalBytes": 73220,
        "intervalDestructiveGetTotalCount": 234,
        "intervalMqputMqput1TotalBytes": 57724,
        "intervalMqputMqput1TotalCount": 149,
        "intervalTopicPutTotal": 6136,
        "logCurrentPrimarySpaceInUsePercentage": 6.52,
        "logFileSystemInUseBytes": 34714415104,
        "logFileSystemMaxBytes": 205019979776,
        "logInUseBytes": 50331648,
        "logLogicalWrittenBytes": 0,
        "logMaxBytes": 83886080,
        "logPhysicalWrittenBytes": 0,
        "logWorkloadPrimarySpaceUtilizationPercentage": 6.52,
        "logWriteLatencySeconds": 0.00054,
        "logWriteSizeBytes": 7338,
        "mqErrorsFileSystemFreeSpacePercentage": 83.06,
        "mqErrorsFileSystemInUseBytes": 34714157056,
        "mqFdcFileCount": 2,
        "mqTraceFileSystemFreeSpacePercentage": 83.06,
        "mqTraceFileSystemInUseBytes": 34714157056,
        "mqcbCount": 0,
        "mqcloseCount": 69,
        "mqconnMqconnxCount": 0,
        "mqctlCount": 0,
        "mqdiscCount": 0,
        "mqinqCount": 32,
        "mqopenCount": 69,
        "mqsetCount": 0,
        "mqstatCount": 0,
        "mqsubrqCount": 0,
        "nonDurableSubscriberHighWaterMark": 46,
        "nonDurableSubscriberLowWaterMark": 28,
        "nonPersistentMessageBrowseBytes": 0,
        "nonPersistentMessageBrowseCount": 0,
        "nonPersistentMessageDestructiveGetCount": 234,
        "nonPersistentMessageMqput1Count": 2,
        "nonPersistentMessageMqputCount": 147,
        "nonPersistentTopicMqputMqput1Count": 21,
        "persistentMessageBrowseBytes": 0,
        "persistentMessageBrowseCount": 0,
        "persistentMessageDestructiveGetCount": 0,
        "persistentMessageMqput1Count": 0,
        "persistentMessageMqputCount": 0,
        "persistentTopicMqputMqput1Count": 0,
        "publishedToSubscribersBytes": 9032,
        "publishedToSubscribersMessageCount": 32,
        "purgedQueueCount": 0,
        "putNonPersistentMessagesBytes": 57724,
        "putPersistentMessagesBytes": 0,
        "queueManagerFileSystemFreeSpacePercentage": 83.06,
        "queueManagerFileSystemInUseBytes": 34714157056,
        "ramFreePercentage": 57.03,
        "ramTotalBytes": 3884974080,
        "ramTotalEstimateForQueueManagerBytes": 204472320,
        "resumeDurableSubscriptionCount": 0,
        "rollbackCount": 0,
        "subscriptionDeleteFailureCount": 0,
        "systemCpuTimeEstimateForQueueManagerPercentage": 0.13,
        "systemCpuTimePercentage": 0.32,
        "topicMqputMqput1IntervalTotal": 21,
        "userCpuTimeEstimateForQueueManagerPercentage": 0,
        "userCpuTimePercentage": 0.05
      }
    }`

**Comment by ibmmqmet** [comment]
The qmgr/queueManager objectType can be fixed simply. Just different code, written at different times ...

It is possible, up to a point, to merge the 2 blocks of metrics for queues (and similarly for qmgr). But they are collected on potentially different schedules. Depending on the relationships between the qmgr's publication period (normally 10 seconds), the `json.interval` and the `global.pollInterval` configuration parameters, you might see either or both sets of data for a queue in a given chunk. I'm not certain whether "differently-shaped" objects are a good or bad thing, rather than simply having unprinted objects when one set of metrics is not available. I'd be interested in feedback on that.

**Comment by lavalex** [comment]
Thank you for the reply.

Yes, I saw that these messages come from a different source in MQ ... I gave the QM example, as it the most obvious one, but my real problem is queue related messages.

I'm sending this messages to elasticsearch for monitoring and alerting. Processing of one huge JSON is very problematic, especially with arrays that contain duplicate object names, so I modified the code to split one huge JSON to several small messages, one per object.

But then, info like current depth of the queue and maximum depth of the queue, ended up in different JSON files and such data correlation in ES is not ... let's say optimal. If only I had this information in the same message ....
